src/mame/sega/segaai.cpp

```diff
@@ -0,0 +1,822 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol, Fabio Priuli
+// thanks-to:Chris Covell
+/*
+
+ Sega AI driver
+
+
+ Not much is known at this stage, except that the system was intended to be
+ used for educational purposes in schools. Yet the audio chips seem much more
+ powerful than what an educational computer requires...
+
+ CPU : 16bit V20 @ 5MHz
+ ROM : 128KB OS.with SEGA PROLOG
+ RAM : 128KB
+ VRAM : 64KB
+ Video : V9938 Resolution 256x212
+ Sound : SN76489
+ Cassette Drive : 9600bps
+ TV Output : RGB, Video, RF
+ Keyboard : new JIS arrangement (Japanese input mapping)
+
+
+TODO:
+- The artwork system has no support for a real touchpad device with
+  selectable artwork, so the touchpad is emulated as a 24x20 matrix
+  of clickable buttons. This is currently good enough to make most
+  games playable. Eventually this should behave like a real touchpad
+  so also drawing apps can work.
```

Comment:
Well, you can use Sega BEENA for inspiration…

Reply:
@wilbertpol See commit https://github.com/mamedev/mame/commit/6e60af32c1e2c2a91937d9184778dad97182f36e

As pointed out there, pen position support isn't yet integrated with the book display, so might be worth waiting for that to be implemented. I can help adapting these changes to Sega AI if needed.

Reply:
> As pointed out there, pen position support isn't yet integrated with the book display, so might be worth waiting for that to be implemented.

Did you miss 585a0f6f47613ed6d82754ef82faf37fc8be457d?  It needs better scans to line up properly, but it’s already usable.

Reply:
I did miss it, wasn't tagged on that one.

Reply:
Thanks, I will take a look.

Reply:
I changed to something similar to what is used for the Beena.
I do not know why but it is not working for the Sega AI yet. I must be missing something somewhere. The cursor is not showing up; however, if I make a similar change to the layout of a different system then it just works and the cursor gets displayed. I will have to investigate this further.